projects/github-actions/repo-gardening/src/get-next-valid-milestone.js

```diff
@@ -28,15 +28,19 @@ async function getNextValidMilestone( octokit, owner, repo, plugin = 'jetpack' )
 	const responses = octokit.paginate.iterator( options );
 
 	for await ( const response of responses ) {
-		// Find a milestone which name is a version number
-		// and it's due dates is earliest in a future
+		// Find a milestone which name is {plugin}/version number
+		// and its due date is earliest in a future.
+		// If we cannot find a milestone with a due date set,
+		// fallback to one without a due date.
 		const reg = new RegExp( '^' + plugin + '\\/\\d+\\.\\d' );
 		const nextMilestone = response.data
 			.filter( m => m.title.match( reg ) )
 			.sort( ( m1, m2 ) =>
 				compareVersions( m1.title.split( '/' )[ 1 ], m2.title.split( '/' )[ 1 ] )
 			)
-			.find( milestone => milestone.due_on && moment( milestone.due_on ) > moment() );
+			.find(
+				milestone => ( milestone.due_on && moment( milestone.due_on ) > moment() ) || milestone
```

Comment:
I don't think this will do the right thing. Since `( milestone.due_on && moment( milestone.due_on ) > moment() ) || milestone` is always going to be truthy, it'll just return the first milestone.

We probably need something like this
```js
// Find all valid milestones for the specified plugin.
const reg = new RegExp( '^' + plugin + '\\/\\d+\\.\\d' );
const milestones = response.data
    .filter( m => m.title.match( reg ) )
    .sort( ( m1, m2 ) => ... );

// Return the first milestone with a future due date, or failing that the first milestone with no due date.
return milestones.find( milestone => milestone.due_on && moment( milestone.due_on ) > moment() ) || 
    milestones.find( milestone => ! milestone.due_on );
```

Reply:
Yeah, good call. I pushed those changes in ca1ffb8afe968d5069092caa1af5d63f7b887e67